runelite-client/src/main/java/net/runelite/client/plugins/opponentinfo/OpponentInfoOverlay.java

```diff
@@ -194,11 +197,15 @@ else if (opponent instanceof Player)
 						// so we know nothing about the upper limit except that it can't be higher than maxHealth
 						maxHealth = lastMaxHealth;
 					}
-					// Take the average of min and max possible healts
+					// Take the average of min and max possible healths
 					health = (minHealth + maxHealth + 1) / 2;
 				}
 
-				progressBarComponent.setLabelDisplayMode(ProgressBarComponent.LabelDisplayMode.FULL);
+				// Show both the hitpoint and percentage values if enabled in the config
+				final ProgressBarComponent.LabelDisplayMode progressBarDisplayMode = displayStyle == HitpointsDisplayStyle.BOTH ?
+					ProgressBarComponent.LabelDisplayMode.BOTH : ProgressBarComponent.LabelDisplayMode.FULL;
+
+				progressBarComponent.setLabelDisplayMode(progressBarDisplayMode);
```

Comment:
I think `setLabelDisplayMode` is missing on the other side of the branch

Reply:
Do you want an explicit call to `setLabelDisplayMode(LabelDisplayMode.PERCENT)` for when the maxHP isn't known/percentage requested in the config?

It's currently the default value for a newly constructed progress bar, which happens on line 163.

Reply:
Oh, I had misread that. This seems fine.